DOCUMENTATION.md

```diff
@@ -0,0 +1,30 @@
+This document describes our rules for documenting things
+
+## The Rules
+
+1. Remember that all our documentation is public-facing.
+2. All engineering documentation in its final state must live in GitHub.
+3. Draft / WIP engineering documentation can live in Google Drive if desired
+4. Code documenation of 3 paragraphs or longer must go into a `.md` file in the same location as the code. You must be able to find every engineering `.md` file via "walking down" links from the root README.md of the repository. Which is to say, add a link from `DEVELOPEMENT.md` or `OPERATIONS.md` or `README.md` (any of them, there are several) into your new `.md` file.
+6. Code comments of 1 paragraph of longer, [or otherwise sufficiently complex](https://github.com/HHS/simpler-grants-gov/blob/02d99c055734edb3d705044de669f7856c0a8fe7/api/tests/src/api/users/test_user_route_login.py#L16-L41), must be linked to from a `.md` file, any `.md` file, pick whichever one makes the most sense.
```

Comment:
I'm going to move this shortly, there isn't a reason to include it here. Auth was getting built quickly

Reply:
Yeah that's reasonable, sorry if it felt like a callout 😅. I should come up with a mock example for "sufficiently complex" or maybe leave it ambiguous. Ambiguous is fine sometimes.

Reply:
I actually appreciate having a specific example. @coilysiren feel free to use one of my more verbose docstrings as an example. Maybe [this one](https://github.com/HHS/simpler-grants-gov/blob/02d99c055734edb3d705044de669f7856c0a8fe7/analytics/src/analytics/metrics/utils.py#L111-L118)?

I struggle because (speaking of colocation) it can be helpful to have the steps outlined directly in the method docstrings, but I also have a more detailed example in [`documentation/analytics/metrics/burndown.md`](https://github.com/HHS/simpler-grants-gov/blob/02d99c055734edb3d705044de669f7856c0a8fe7/documentation/analytics/metrics/burndown.md#step-by-step-example-sprint-burndown)

In the case of the examples linked above, would you recommend removing the steps listed under `Notes` in favor of a link to that markdown file (after moving it into the `analytics/` codebase)?